gems/sorbet-runtime/test/types/props/serializable.rb

```diff
@@ -494,26 +494,275 @@ class CustomTypeStruct < T::Struct
+      assert_equal(:hash_key, storytime[:prop])
+      assert_equal(obj, storytime[:value])
+    end
+
     it 'round trips as hash value' do
       assert_equal({'foo' => 'bar'}, CustomTypeStruct.from_hash({'hash_value' => {'foo' => 'bar'}}).serialize['hash_value'])
     end
 
+    it 'raises serialize errors for members needing transform_values' do
+      obj = CustomType.new
+      struct = CustomTypeStruct.new
+      struct.instance_variable_set(:@hash_value, obj)
+      e = assert_raises(NoMethodError) do
+        struct.serialize
+      end
+
+      assert_includes(e.message, "undefined method `transform_values'")
+    end
+
+    it 'raises deserialize errors for members needing transform_values' do
+      msg_string = nil
+      extra_hash = nil
+      T::Configuration.soft_assert_handler = proc do |msg, extra|
+        msg_string = msg
+        extra_hash = extra
+      end
+
+      obj = 'not a hash'
+      result = CustomTypeStruct.from_hash({'hash_value' => obj})
+      assert_equal('not a hash', result.hash_value)
+
+      refute_nil(msg_string)
+      refute_nil(extra_hash)
+      storytime = extra_hash[:storytime]
+      assert_equal(CustomTypeStruct, storytime[:klass])
+      assert_equal(:hash_value, storytime[:prop])
+      assert_equal(obj, storytime[:value])
+    end
+
     it 'round trips as hash key and value' do
       assert_equal({'foo' => 'bar'}, CustomTypeStruct.from_hash({'hash_both' => {'foo' => 'bar'}}).serialize['hash_both'])
     end
+
+    it 'raises serialize errors for members needing each_with_object' do
+      obj = CustomType.new
+      struct = CustomTypeStruct.new
+      struct.instance_variable_set(:@hash_both, obj)
+      e = assert_raises(NoMethodError) do
+        struct.serialize
+      end
+
+      assert_includes(e.message, "undefined method `each_with_object'")
+    end
+
+    it 'raises deserialize errors for members needing each_with_object' do
+      msg_string = nil
+      extra_hash = nil
+      T::Configuration.soft_assert_handler = proc do |msg, extra|
+        msg_string = msg
+        extra_hash = extra
+      end
+
+      obj = 'not a hash'
+      result = CustomTypeStruct.from_hash({'hash_both' => obj})
+      assert_equal('not a hash', result.hash_both)
+
+      refute_nil(msg_string)
+      refute_nil(extra_hash)
+      storytime = extra_hash[:storytime]
+      assert_equal(CustomTypeStruct, storytime[:klass])
+      assert_equal(:hash_both, storytime[:prop])
+      assert_equal(obj, storytime[:value])
+    end
+  end
+
+  class SetPropStruct
+    include T::Props::Serializable
+    prop :set, T::Set[String]
+  end
+
+  describe 'set props' do
+    it 'round trips' do
+      set = Set['foo']
+      struct = SetPropStruct.new
+      struct.set = set
+
+      h = struct.serialize
+      assert_equal(1, h['set'].length)
+      assert(h['set'].include?('foo'))
+
+      roundtripped = SetPropStruct.from_hash(h)
+      assert_equal(1, roundtripped.set.length)
```

Comment:
Could we just `assert_equal(roundtripped.set, set)`?

Reply:
Ah, yeah, that'd be easier, wouldn't it...